cocotb/binary.py

```diff
@@ -91,29 +96,42 @@ class BinaryValue(object):
     '*'
 
     """
+    # Accepted characters and resolution
     _resolve_to_0     = "-lL"  # noqa
     _resolve_to_1     = "hH"  # noqa
     _resolve_to_error = "xXzZuUwW"  # Resolve to a ValueError() since these usually mean something is wrong
     _permitted_chars  = _resolve_to_0 +_resolve_to_1 + _resolve_to_error + "01"  # noqa
 
     def __init__(self, value=None, n_bits=None, bigEndian=True,
                  binaryRepresentation=BinaryRepresentation.UNSIGNED,
-                 bits=None):
+                 bits=None, ascending_range=False):
         """Args:
             value (str or int or long, optional): Value to assign to the bus.
             n_bits (int, optional): Number of bits to use for the underlying
                 binary representation.
-            bigEndian (bool, optional): Interpret the binary as big-endian
-                when converting to/from a string buffer.
+            bigEndian (bool, optional): Interpret the binary string as
+                big-endian when converting to/from a byte string buffer.
+                AKA byte-ordering
             binaryRepresentation (BinaryRepresentation): The representation
                 of the binary value
                 (one of :any:`UNSIGNED`, :any:`SIGNED_MAGNITUDE`, :any:`TWOS_COMPLEMENT`).
                 Defaults to unsigned representation.
+            ascending_range (bool, optional): Interpret the binary string bit-order
+                as having an ascending range (big-endian bits).
             bits (int, optional): Deprecated: Compatibility wrapper for :attr:`n_bits`.
         """
+        # The underlying vector:
+        # - Mirrors the representation in HW
+        # - This is a Python string of bits
+        # - Note that because it is a string, if acending_range is false, indexing (of strings) no
+        #   longer provides the expected result (0th index is still the start of the string)
         self._str = ""
+
         self.big_endian = bigEndian
         self.binaryRepresentation = binaryRepresentation
+        self.ascending_range = ascending_range
+
+        self._log = SimLog("cocotb.binary.%s" % self.__class__.__name__)
```

Comment:
Having a log lookup per binary value might be massively expensive. Can you make this a class-level member?

Reply:
Is that as simple as moving it out of the constructor as `_log = SimLog("cocotb.binary.%s" % BinaryValue.__class__.__name__)` ?

Reply:
Yes - but `BinaryValue.__class__.__name__` is a weird way to spell `"BinaryValue"`

Reply:
Aha yeah, I was thinking it would inherit - but that wouldn't!